File: services/education/internal/model/order_tag.go
package model

import "time"

type OrderTagPagedQueryDTO struct {
	PageRequestModel PageRequestModel          `json:"pageRequestModel"`
	QueryModel       OrderTagPagedQueryFilters `json:"queryModel"`
}

type OrderTagPagedQueryFilters struct {
	Enable *bool `json:"enable"`
}

type OrderTagManageVO struct {
	ID            string     `json:"id"`
	Name          string     `json:"name"`
	Enable        bool       `json:"enable"`
	OrgOrderTagID string     `json:"orgOrderTagId"`
	CreatedTime   *time.Time `json:"createdTime,omitempty"`
	UpdatedTime   *time.Time `json:"updatedTime,omitempty"`
}

type OrderTagPagedResult struct {
	List  []OrderTagManageVO `json:"list"`
	Total int                `json:"total"`
}

type CreateOrderTagDTO struct {
	Name string `json:"name"`
}

type UpdateOrderTagDTO struct {
	ID     int64  `json:"id"`
	Name   string `json:"name"`
	Enable *bool  `json:"enable"`
}

File: platform-admin/src/api/finance-center/order-tag.ts
import { usePost } from '~/utils/request'

export interface OrderTagItem {
  id: string
  name: string
  enable: boolean
  orgOrderTagId: string
  createdTime?: string
  updatedTime?: string
}

export interface OrderTagPagedResult {
  list?: OrderTagItem[]
  total?: number
}

export function getOrderTagListPagedApi(data: {
  queryModel?: {
    enable?: boolean
  }
  sortModel?: Record<string, any>
  pageRequestModel: {
    needTotal?: boolean
    pageSize: number
    pageIndex: number
    skipCount?: number
  }
}) {
  return usePost<OrderTagPagedResult>('/api/v1/order-tags/list-paged', data)
}

export function createOrderTagApi(data: {
  name: string
}) {
  return usePost<{ id: number }>('/api/v1/order-tags/create', data)
}

export function updateOrderTagApi(data: {
  id: string | number
  name?: string
  enable?: boolean
}) {
  return usePost<boolean>('/api/v1/order-tags/update', data)
}
